### ISSUE TITLE
Fix deprecated api call in _NonAtrousConvolution

### ISSUE THREAD

**Issue by se4u** [tensorflow/tensorflow#17842]
### System information
I am using tf version `1.7.0-dev20180317, v1.6.0-rc1-1580-gc941c087a9` on OSX in standard non-eager mode.

### Describe the problem
Lines `151-156` in class `_NonAtrousConvolution` in `tensorflow/python/ops/nn_ops.py` set `nhwc` as data format when `conv_dims` is 1.  

``` python
    151     if conv_dims == 1:
    152       # conv1d uses the 2-d data format names
    153       if data_format is None or data_format == "NWC":
    154         data_format_2d = "NHWC"
    155       elif data_format == "NCW":
    156         data_format_2d = "NCHW"
```

But this causes a warning later on at line `2384` when `conv1d` is called from inside the `_conv1d` method at line `189`

```
WARNING:tensorflow:From ~/tensorflow/python/util/deprecation.py:497: calling conv1d (from tensorflow.python.ops.nn_ops) with data_format=NHWC is deprecated and will be removed in a future version.
Instructions for updating:
`NHWC` for data_format is deprecated, use `NWC` instead
```

**Comment by godspeed1989** [comment]
I encounter the same warning with tf  version `1.6.0` on ArchLinux
```
WARNING:tensorflow:From /usr/lib/python3.6/site-packages/tensorflow/python/util/deprecation.py:497: calling conv1d (from tensorflow.python.ops.nn_ops) with data_format=NHWC is deprecated and will be removed in a future version.
Instructions for updating:
`NHWC` for data_format is deprecated, use `NWC` instead
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by imsheridan** [comment]
This issue is assumed to be fixed by #17455

**Comment by se4u** [comment]
Closing issue since #17455 was merged into master so it will be available in the nightly builds even if not in 1.7
